### PR TITLE
XMLHttpRequest: upload listener flag

### PR DESCRIPTION
**Pull request by annevk**
Tests for https://github.com/whatwg/xhr/issues/95.

### CODE REVIEW

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5122/)

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/210554335)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 6426c39afa34414d85532b1f7dfbb9d301a92c45
Using browser at version BuildID 20170312110221; SourceStamp 3154da520af568d4da3de62bc26fbf97a1317f51
Starting 10 test iterations
## Unstable results ##


|                           Test                          |                                                                         Subtest                                                                          |          Results           |                   Messages                   |
|---------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------|----------------------------------------------|
| `/XMLHttpRequest/event-upload-progress-crossorigin.htm` | `Upload events registered too late (resources/redirect.py?code=307&location=http://www1.web-platform.test:8000/XMLHttpRequest/resources/corsenabled.py)` | **FAIL: 4/10, PASS: 6/10** | `assert_unreached: Reached unreachable code` |

## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5122/XMLHttpRequest/event-upload-progress-crossorigin.htm">/XMLHttpRequest/event-upload-progress-crossorigin.htm</a></summary>


|                                                                         Subtest                                                                          |          Results           |                   Messages                   |
|----------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------|----------------------------------------------|
|                                                                                                                                                          | OK                         |                                              |
| `Upload events registered on time (http://www1.web-platform.test:8000/XMLHttpRequest/resources/corsenabled.py)`                                          | PASS                       |                                              |
| `Upload events registered on time (resources/redirect.py?code=307&location=http://www1.web-platform.test:8000/XMLHttpRequest/resources/corsenabled.py)`  | PASS                       |                                              |
| `Upload events registered too late (http://www1.web-platform.test:8000/XMLHttpRequest/resources/corsenabled.py)`                                         | PASS                       |                                              |
| `Upload events registered too late (resources/redirect.py?code=307&location=http://www1.web-platform.test:8000/XMLHttpRequest/resources/corsenabled.py)` | **FAIL: 4/10, PASS: 6/10** | `assert_unreached: Reached unreachable code` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/210554336)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 6426c39afa34414d85532b1f7dfbb9d301a92c45
Using browser at version 58.0.3029.14 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5122/XMLHttpRequest/event-upload-progress-crossorigin.htm">/XMLHttpRequest/event-upload-progress-crossorigin.htm</a></summary>


|                                                                         Subtest                                                                          | Results |                   Messages                   |
|----------------------------------------------------------------------------------------------------------------------------------------------------------|---------|----------------------------------------------|
|                                                                                                                                                          | OK      |                                              |
| `Upload events registered on time (http://www1.web-platform.test:8000/XMLHttpRequest/resources/corsenabled.py)`                                          | PASS    |                                              |
| `Upload events registered on time (resources/redirect.py?code=307&location=http://www1.web-platform.test:8000/XMLHttpRequest/resources/corsenabled.py)`  | FAIL    | `assert_unreached: Reached unreachable code` |
| `Upload events registered too late (http://www1.web-platform.test:8000/XMLHttpRequest/resources/corsenabled.py)`                                         | PASS    |                                              |
| `Upload events registered too late (resources/redirect.py?code=307&location=http://www1.web-platform.test:8000/XMLHttpRequest/resources/corsenabled.py)` | PASS    |                                              |

</details>
</details>

**Comment by annevk** [comment]
@wisniewskit could you investigate this? It's somewhat flaky in Firefox. After enough refreshing it passes all tests, which seems somewhat absurd.

**Comment by wisniewskit** [comment]
I'll try to take a look ASAP (which may not be this week).

**Comment by annevk** [comment]
@wisniewskit ping.

**Comment by annevk** [comment]
@wisniewskit ping.

**Comment by annevk** [comment]
@ricea maybe you can take a look at this?

**Comment by ricea** [comment]
Looking ...

File: xhr/event-upload-progress.htm [hunk 1]
```diff
@@ -4,25 +4,30 @@
...
-      assert_equals(e.total, 22);
-      test.done();
-    });
-    client.onreadystatechange = test.step_func(function() {
-      if (client.readyState === 4) assert_unreached("onprogress not called.");
-    });
-    client.open("POST", "resources/upload.py");
-    client.send("This is a test string.");
-  });
+const remote = get_host_info().HTTP_REMOTE_ORIGIN + "/XMLHttpRequest/resources/corsenabled.py",
```
Comment (ricea) [inline xhr/event-upload-progress.htm]: I think /XMLHttpRequest/ should be /xhr/ here.

Comment (annevk) [reply xhr/event-upload-progress.htm]: Thanks, I must have rebased this after the rename and forgotten about this or some such.

**Review by ricea — APPROVED** [review]

**Comment by annevk** [comment]
Thanks! (Browser bugs are linked from the XMLHttpRequest issue.)

**Pull request merged**